packages/ckeditor5-table/src/tablelayout/tablelayoutediting.ts

```diff
@@ -0,0 +1,212 @@
+
+	/**
+	 * Defines the converters for the table layout feature.
+	 */
+	private _defineConverters() {
+		const { editor } = this;
+		const { conversion } = editor;
+
+		conversion.for( 'upcast' ).add( upcastTableOverride() );
+		conversion.for( 'upcast' ).add( upcastTableToSetTableType() );
+
+		editor.conversion.for( 'dataDowncast' ).add( dispatcher => {
+			return dispatcher.on( 'attribute:tableType:table', ( evt, data, conversionApi ) => {
+				const { item, attributeNewValue } = data;
+				const { mapper, writer } = conversionApi;
+
+				if ( !conversionApi.consumable.test( item, evt.name ) ) {
+					return;
+				}
+
+				const table = mapper.toViewElement( item );
+
+				writer.addClass( `${ attributeNewValue }-table`, table );
+				writer.setAttribute( 'role', 'presentation', table );
+
+				conversionApi.consumable.consume( item, evt.name );
+			} );
+		} );
+
+		conversion.for( 'editingDowncast' ).add( dispatcher => {
+			dispatcher.on( 'attribute:tableType', ( evt, data, conversionApi ) => {
+				const { item, attributeNewValue } = data;
+				const { mapper, writer } = conversionApi;
+				const modelElement = item;
+				const viewElement = mapper.toViewElement( modelElement );
+
+				writer.addClass( `${ attributeNewValue }-table`, viewElement );
+			} );
+		} );
+	}
+}
+
+/**
+ * View table element to model table element conversion helper.
+ *
+ * This conversion helper overrides the default table converter to meet table layout conditions.
+ *
+ * @returns Conversion helper.
+ */
+function upcastTableOverride() {
+	return ( dispatcher: UpcastDispatcher ): void => {
+		dispatcher.on<UpcastElementEvent>( 'element:table', (
+			evt: EventInfo,
+			data: UpcastConversionData<ViewElement>,
+			conversionApi: UpcastConversionApi
+		) => {
+			const viewTable = data.viewItem;
+
+			// When element was already consumed then skip it.
+			if ( !conversionApi.consumable.test( viewTable, { name: true } ) ) {
+				return;
+			}
+
+			const { rows, headingRows, headingColumns } = scanTable( viewTable );
+
+			// Only set attributes if values are greater then 0.
+			const attributes: { headingColumns?: number; headingRows?: number } = {};
+
+			const hasTableTypeContent = isTableTypeContent( viewTable );
+
+			if ( headingColumns && hasTableTypeContent ) {
+				attributes.headingColumns = headingColumns;
+			}
+
+			if ( headingRows && hasTableTypeContent ) {
+				attributes.headingRows = headingRows;
+			}
+
+			const table = conversionApi.writer.createElement( 'table', attributes );
+
+			if ( !conversionApi.safeInsert( table, data.modelCursor ) ) {
+				return;
+			}
+
+			conversionApi.consumable.consume( viewTable, { name: true } );
+
+			// Upcast table rows in proper order (heading rows first).
+			rows.forEach( row => conversionApi.convertItem( row, conversionApi.writer.createPositionAt( table, 'end' ) ) );
+
+			// Convert everything else.
+			conversionApi.convertChildren( viewTable, conversionApi.writer.createPositionAt( table, 'end' ) );
```

Comment:
Please check how table columns are upcasted so they are at the end of table as model expects.